### PR TITLE
Fix issue #138

### PR DESCRIPTION
**Pull request by ToniRV**

### CODE REVIEW

File: src/KimeraVioRos.cpp [hunk 1]
```diff
@@ -121,15 +121,15 @@ bool KimeraVioRos::spin() {
...
     std::future<bool> vio_viz_handle =
         std::async(std::launch::async,
                    &VIO::Pipeline::spinViz,
-                   std::ref(*CHECK_NOTNULL(vio_pipeline_.get())));
+                   CHECK_NOTNULL(vio_pipeline_.get()));
     std::future<bool> vio_pipeline_handle =
         std::async(std::launch::async,
                    &VIO::Pipeline::spin,
-                   std::ref(*CHECK_NOTNULL(vio_pipeline_.get())));
+                   *CHECK_NOTNULL(vio_pipeline_.get()));
```
Comment (lucacarlone) [inline src/KimeraVioRos.cpp:132]: lgtm: why do you need "*" here?

**Review by lucacarlone — APPROVED** [review]

File: src/KimeraVioRos.cpp (hunk 1 above)
Comment (ToniRV) [reply src/KimeraVioRos.cpp:132]: `CHECK_NOTNULL(x)` returns the pointer `x` (if not null), so you still need to dereference it.

Comment (lucacarlone) [reply src/KimeraVioRos.cpp:132]: got it thanks. I'll merge after Marcus reviews

**Review by marcusabate — APPROVED** [review]
Runs on my machine which is 20.04. I think there is a dockerfile for 18.04 on the private side, would help to merge that here so that we can reproduce builds on specific OSes.

**Pull request merged**